File: my-folder/0452-minimum-number-of-arrows-to-burst-balloons/solution.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        int arrow = 1;
        int start = points[0][0];
        int end = points[0][1];

        for (int i = 1; i < points.length; i++) {
            int xStart = points[i][0];
            int xEnd = points[i][1];

            if (xStart <= end) {
                start = xStart;
            }
            else {
                arrow++;
                start = xStart;
                end = xEnd;
            }
        }
        return arrow;
    }
}

File: my-folder/0452-minimum-number-of-arrows-to-burst-balloons/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points: return 0

        sorted_arr = sorted(points, key = lambda x: x[1])

        count = 1

        end_point = sorted_arr[0][1]

        for start, end in sorted_arr[1:]:
            if start > end_point:
                end_point = end
                count += 1
        
        return count
